packages/core/src/codecatalyst/auth.ts

```diff
@@ -379,10 +379,12 @@ export class CodeCatalystAuthenticationProvider {
         if (isValidCodeCatalystConnection(conn)) {
             getLogger().info(`auth: re-use connection from existing connection id ${connId}`)
             await this.secondaryAuth.useNewConnection(conn)
+            await this.isConnectionOnboarded(conn, true)
```

Comment:
It seems non-obvious that `tryUseConnection` is doing this? Is there a 1-line comment that can give a hint . Also for the one below. 

```suggestion
            // HACK: 
            await this.isConnectionOnboarded(conn, true)
```